test/sol6/kyberStorage.js

```diff
@@ -501,92 +501,127 @@ contract('KyberStorage', function(accounts) {
+            // 2 indices
+            result = await kyberStorage.getReserveAddressesPerTokenSrc(token.address, 0, 1);
+            Helper.assertEqual(result.length, 2, "T2E should be listed");
+            Helper.assertEqual(result[0], reserveAddresses[0], "T2E should be listed");
+            Helper.assertEqual(result[1], reserveAddresses[1], "T2E should be listed");
+            // end index is big
+            result = await kyberStorage.getReserveAddressesPerTokenSrc(token.address, 0, 3);
+            Helper.assertEqual(result.length, 2, "T2E should be listed");
+            Helper.assertEqual(result[0], reserveAddresses[0], "T2E should be listed");
+            Helper.assertEqual(result[1], reserveAddresses[1], "T2E should be listed");
+            // start + end indices are big
+            result = await kyberStorage.getReserveAddressesPerTokenSrc(token.address, 3, 4);
+            Helper.assertEqual(result.length, zeroBN);
+            // start index > end index
+            result = await kyberStorage.getReserveAddressesPerTokenSrc(token.address, 1, 0);
+            Helper.assertEqual(result.length, zeroBN);
+            // delist for both side
+            for (let reserve of Object.values(reserveInstances)) {
+                await kyberStorage.listPairForReserve(reserve.address, token.address, true, true, false, {from: operator});
+            }
+        });
+
+        it("should list E2T side with 2 reserve", async() => {
             let reserveIds = [];
             for (let reserve of Object.values(reserveInstances)) {
                 await kyberStorage.listPairForReserve(reserve.address, token.address, true, false, true, {from: operator});
                 reserveIds.push(reserve.reserveId);
             }
             let result = await kyberStorage.getReservesPerTokenSrc(token.address);
-            Helper.assertEqual(result.length, zeroBN, "E2T should not be listed");
-            result = await kyberStorage.getReserveAddressesPerTokenSrc(token.address);
-            Helper.assertEqual(result.length, zeroBN, "E2T should not be listed");
+            Helper.assertEqual(result.length, zeroBN, "T2E should not be listed");
+            result = await kyberStorage.getReserveAddressesPerTokenSrc(token.address, 0, 1);
+            Helper.assertEqual(result.length, zeroBN, "T2E should not be listed");
 
             result = await kyberStorage.getReservesPerTokenDest(token.address);
-            Helper.assertEqualArray(result, reserveIds, "T2E should be listed");
+            Helper.assertEqualArray(result, reserveIds, "E2T should be listed");
             // delist for both side
             for (let reserve of Object.values(reserveInstances)) {
                 await kyberStorage.listPairForReserve(reserve.address, token.address, true, true, false, {from: operator});
             }
         });
 
-        it("should list T2E side only", async() => {
+        it("should list E2T side only", async() => {
```

Comment:
why these changes?

Reply:
The listing params: `(reserve, token, eth2Token, token2Eth, add)`
so here we use: `(reserve, token, true, false, true)` means E2T is listed only right?

Reply:
yes
Got it.